app/src/main/java/org/kiwix/kiwixmobile/language/viewmodel/SaveLanguagesAndFinish.kt

```diff
@@ -33,7 +33,7 @@ data class SaveLanguagesAndFinish(
 
   override fun invokeWith(activity: AppCompatActivity) {
     Flowable.fromCallable { languageDao.insert(languages) }
-      .subscribeOn(Schedulers.io())
+      .subscribeOn(AndroidSchedulers.mainThread())
```

Comment:
@MohitMaliFtechiz You should subscribe to the io thread as it is doing the database operations and you observe that on the main thread for the UI. 
```
 .subscribeOn(Schedulers.io())
 .observeOn(AndroidSchedulers.mainThread())
```

Reply:
@gouri-panda ,
done